OSDOCS-20430 created RNs for zero-trust 1.1.0 (#114294)
openshift/openshift-docs: modules/zero-trust-manager-release-notes-1-1-0.adoc

The following advisories are available for the {zero-trust-full}:

* https://access.redhat.com/errata/RHBA-2026:28869[RHBA-2026:28869]
* https://access.redhat.com/errata/RHBA-2026:28784[RHBA-2026:28784]
* https://access.redhat.com/errata/RHBA-2026:28399[RHBA-2026:28399]
* https://access.redhat.com/errata/RHBA-2026:28392[RHBA-2026:28392]
* https://access.redhat.com/errata/RHBA-2026:28391[RHBA-2026:28391]
* https://access.redhat.com/errata/RHBA-2026:28260[RHBA-2026:28260]
* https://access.redhat.com/errata/RHBA-2026:28200[RHBA-2026:28200]
* https://access.redhat.com/errata/RHBA-2026:28140[RHBA-2026:28140]

{zero-trust-full} supports the following components and versions:

[cols="1,1",options="header"]
|===
| Component
| Version

> | SPIRE Server

Reviewer: Upstream versions are incorrect. Please correct them.
I see that many of them have incorrect version in this PR.

Reviewer 2: @wgabor0427 Have the incorrect versions been corrected?

Author: Versions are correct

| 1.13.3

| SPIRE Agent
| 1.13.3

| SPIRE Controller Manager
| 0.6.3

| SPIRE OIDC Discovery Provider
| 1.13.3

| SPIFFE CSI Driver
| 0.2.8
|===

[id="zero-trust-manager-1-1-0-features-enhancements_{context}"]
== New features and enhancements

Supported SPIFFE Helper container image::

{zero-trust-full} now provides a supported SPIFFE Helper container image for workloads that cannot use the SPIFFE Workload API directly but can read Transport Layer Security (TLS) credentials from a shared volume. The image is based on upstream SPIFFE Helper. The configuration file format, command-line flags, and Workload API behavior remain compatible.

SPIRE UpstreamAuthority plugins for external certificate authorities::

> {zero-trust-full} now documents SPIRE Server UpstreamAuthority plugins that obtain intermediate signing certificates from external certificate management systems while preserving {spiffe-full} identity standards.

Reviewer 3: ```suggestion
{zero-trust-full} now supports SPIRE Server UpstreamAuthority plugins that obtain intermediate signing certificates from external certificate management systems while preserving {spiffe-full} identity standards.
```

// Module included in the following assemblies:
//
// * security/zero_trust_workload_identity_manager/zero-trust-manager-release-notes.adoc
:_mod-docs-content-type: REFERENCE
[id="zero-trust-manager-release-notes-1-1-0_{context}"]
= {zero-trust-full} 1.1.0

Issued: 30 Jun 2026

[role="_abstract"]

> This release adds integration and operational capabilities for workloads that use external certificate authorities, service mesh deployments, or file-based Transport Layer Security (TLS) credentials. The release includes a supported SPIFFE Helper container image, SPIRE UpstreamAuthority plugins for cert-manager and HashiCorp Vault, and {SMProductName} integration with SPIRE for single-cluster and federated multi-cluster mutual Transport Layer Security (mTLS).

Reviewer 2: Should "UpstreamAuthority" be in backticks?

* https://access.redhat.com/errata/RHBA-2026:28391[RHBA-2026:28391]
* https://access.redhat.com/errata/RHBA-2026:28260[RHBA-2026:28260]
* https://access.redhat.com/errata/RHBA-2026:28200[RHBA-2026:28200]
* https://access.redhat.com/errata/RHBA-2026:28140[RHBA-2026:28140]

{zero-trust-full} supports the following components and versions:

[cols="1,1",options="header"]
|===
| Component
| Version

| {zero-trust-full}
| 1.1.0

| SPIRE Server
| 1.14.7

| SPIRE Agent
| 1.14.7

| SPIRE Controller Manager
| 0.6.4

| SPIRE OIDC Discovery Provider
| 1.14.7

| SPIFFE CSI Driver
| 0.2.8
|===

[id="zero-trust-manager-1-1-0-features-enhancements_{context}"]
== New features and enhancements

Supported SPIFFE Helper container image::

{zero-trust-full} now provides a supported SPIFFE Helper container image for workloads that cannot use the SPIFFE Workload API directly but can read Transport Layer Security (TLS) credentials from a shared volume. The image is based on upstream SPIFFE Helper. The configuration file format, command-line flags, and Workload API behavior remain compatible.

SPIRE UpstreamAuthority plugins for external certificate authorities::

> {zero-trust-full} now supports SPIRE Server UpstreamAuthority plugins that obtain intermediate signing certificates from external certificate management systems while preserving {spiffe-full} identity standards.

Reviewer 2: Should "UpstreamAuthority" be in backticks?

Author: no back ticks for UpstreamAuthority

* Supported plugins:

> 

Reviewer 2: Same question about "UpstreamAuthority" on lines 63 and 64.

Author: Same

> ** cert-manager UpstreamAuthority plugin: integrates SPIRE Server with {cert-manager-operator}

Reviewer 2: ```suggestion
** cert-manager UpstreamAuthority plugin: integrates SPIRE Server with {cert-manager-operator}.
```

Author: Done

> ** Vault UpstreamAuthority plugin: integrates SPIRE Server with the HashiCorp Vault Public Key Infrastructure (PKI) secrets engine

Reviewer 2: ```suggestion
** Vault UpstreamAuthority plugin: integrates SPIRE Server with the HashiCorp Vault Public Key Infrastructure (PKI) secrets engine.
```

Author: Done

cert-manager UpstreamAuthority plugin::

> The cert-manager UpstreamAuthority plugin connects SPIRE Server to {cert-manager-operator} for automated intermediate certificate provisioning. SPIRE Server creates a `CertificateRequest` custom resource, then the configured Issuer or ClusterIssuer signs the request, and then the SPIRE Server uses the signed intermediate certificate to issue workload identities.

Reviewer 2: ```suggestion
The cert-manager UpstreamAuthority plugin connects SPIRE Server to {cert-manager-operator} for automated intermediate certificate provisioning. SPIRE Server creates a `CertificateRequest` custom resource, then the configured `Issuer` or `ClusterIssuer` signs the request, and then the SPIRE Server uses the signed intermediate certificate to issue workload identities.
```
Based on what I see in the docs, "Issuer" and "ClusterIssuer" should be in backticks. Should "UpstreamAuthority" also be in backticks?

Author: Issuer and ClusterIssuer yes UpstreamAuthority no

> 

Reviewer 2: Same question about the formatting of "UpstreamAuthority" on lines 70  - 72.

Vault UpstreamAuthority plugin::

> The Vault UpstreamAuthority plugin connects SPIRE Server to the HashiCorp Vault PKI secrets engine for automated intermediate CA certificate signing. Use this plugin when PKI is centralized in Vault and SPIRE should obtain intermediate signing certificates through Vault policies and authentication.

Reviewer 2: ```suggestion
The Vault UpstreamAuthority plugin connects SPIRE Server to the HashiCorp Vault PKI secrets engine for automated intermediate certificate authority (CA) certificate signing. Use this plugin when PKI is centralized in Vault and SPIRE must obtain intermediate signing certificates through Vault policies and authentication.
```

Author: Done

Single-cluster Service Mesh integration with SPIRE::

{zero-trust-full} now supports single-cluster integration with {SMProductName}. SPIRE replaces Istio's built-in certificate authority (CA) with SPIFFE-compliant identities and short-lived certificates that rotate automatically for workload mutual Transport Layer Security (mTLS).

Multi-cluster Service Mesh integration with SPIRE federation::

{zero-trust-full} now supports multi-cluster integration with {SMProductName} through SPIRE federation, enabling cross-cluster mutual Transport Layer Security (mTLS) authentication and zero trust workload identity across separate {product-title} clusters.

* Cross-cluster trust requires federation at two layers:

** SPIRE federation: SPIRE Servers exchange trust bundles through the `https_spiffe` profile.
** Istio federation: Istio discovers remote endpoints through remote secrets and routes traffic through East-West Gateways.

[id="zero-trust-manager-1-1-0-deprecated-features_{context}"]
== Deprecated features

> 

Reviewer 2: I think this section needs to be restructured. If I understand it correctly, you are presenting users with steps to take to handle the deprecated feature. However, the formatting in release-note modules doesn't afford a good way to present what is essentially a procedure.

I suggest creating a short procedure module and moving most of the content from line 98 - 148 there. Then, in this "Deprecated features" section, you can include a short paragraph announcing the deprecation and pointing to the procedure.

Author: I removed the procedure steps

----
+
The expected output is `DaemonSetAvailable=True`, and `Ready=True`.

* Confirm that the CSI `DaemonSet` is available by running the following command:
+
[source,terminal]
----
$ oc get ds spire-spiffe-csi-driver -n zero-trust-workload-identity-manager
----

* Confirm CSI pods are running and using the `privileged` SCC by running the following commands:
+
[source,terminal]
----
$ oc get pods -n zero-trust-workload-identity-manager -l app.kubernetes.io/name=spiffe-csi-driver
----
+
[source,terminal]
----
$ oc get pod -n zero-trust-workload-identity-manager -l app.kubernetes.io/name=spiffe-csi-driver \
     -o jsonpath='{range .items[*]}{.metadata.name}{"\t"}{.metadata.annotations.openshift\.io/scc}{"\n"}{end}'
----
+
Every pod should show `privileged`.

. After all checks have passed, delete the legacy custom SCC by running the following commands:
+
[source,terminal]
----
$ oc get scc spire-spiffe-csi-driver
----
+
[source,terminal]
----
$ oc delete scc spire-spiffe-csi-driver
----
+
If the SCC is already absent, no action is needed.

[id="zero-trust-manager-1-1-0-bug-fixes_{context}"]
== Fixed issues

Managed route TLS secrets no longer require manual RBAC configuration::
+

> * Before this update, when you configured a managed route with an `externalSecretRef` TLS certificate on the `SpireOIDCDiscoveryProvider` or `SpireServer` custom resource, {zero-trust-full} did not create the `RoleBinding` that grants the OpenShift Ingress router service account permission to read the referenced Secret. As a consequence, route reconciliation failed with a `ManagedRouteUpdateFailed` condition and you had to manually create `secret-reader`, `Role`, and `RoleBinding` custom resources (CRs). With this release, {zero-trust-full} automatically reconciles the required `Role` and `RoleBinding` CRs so that the router service account can access the Secrets referenced by the `externalSecretRef`. Managed routes that use externally provided TLS certificates now reconcile without additional manual RBAC steps.

Reviewer 2: ```suggestion
* Before this update, when you configured a managed route with an `externalSecretRef` TLS certificate on the `SpireOIDCDiscoveryProvider` or `SpireServer` custom resource (CR), {zero-trust-full} did not create the `RoleBinding` that grants the OpenShift Ingress router service account permission to read the referenced Secret. As a consequence, route reconciliation failed with a `ManagedRouteUpdateFailed` condition, and you had to manually create `secret-reader`, `Role`, and `RoleBinding` CRs. With this release, {zero-trust-full} automatically reconciles the required `Role` and `RoleBinding` CRs so that the router service account can access the Secrets referenced by the `externalSecretRef` TLS certificate. Managed routes that use externally provided TLS certificates now reconcile without additional manual RBAC steps.
```

Author: Done

+
(link:https://issues.redhat.com/browse/SPIRE-164[SPIRE-164])

Pre-existing operand resources are no longer overwritten at installation::
+
* Before this update, when you installed {zero-trust-full} on a cluster that already contained Kubernetes resources with the same names as operand objects, {zero-trust-full} reconciled and overwrote those resources during the initial installation without reporting a conflict. As a consequence, manually created or third-party resources could be modified unexpectedly before you had a chance to resolve naming collisions. With this release, {zero-trust-full} checks the `app.kubernetes.io/managed-by` label before updating any existing resource during installation. If a matching resource is not managed by {zero-trust-full}, {zero-trust-full} sets a `ResourceConflict` status condition and stops reconciliation instead of overwriting the resource.
+
(link:https://issues.redhat.com/browse/SPIRE-340[SPIRE-340])

Unmanaged cluster resources are protected during reconciliation::
+

> * Before this update, after {zero-trust-full} was already running, operand controllers could still update Kubernetes resources with matching names even when those resources were not owned by {zero-trust-full}. This could occur when a name collision appeared later or when the `app.kubernetes.io/managed-by: zero-trust-workload-identity-manager` label was removed from a resource that {zero-trust-full} had previously managed. With this release, each operand controller verifies the managed-by label on every reconcile cycle before applying updates. When the label is absent, {zero-trust-full} sets a `ResourceConflict` status condition on the custom resource and skips the update to the conflicting object. Operand updates proceed only for resources that {zero-trust-full} currently manages.

Reviewer 2: ```suggestion
* Before this update, after {zero-trust-full} was already running, operand controllers could still update Kubernetes resources with matching names even when those resources were not owned by {zero-trust-full}. This could occur when a name collision appeared later or when the `app.kubernetes.io/managed-by: zero-trust-workload-identity-manager` label was removed from a resource that {zero-trust-full} had previously managed. With this release, each operand controller verifies the `managed-by` label on every reconcile cycle before applying updates. When the label is absent, {zero-trust-full} sets a `ResourceConflict` status condition on the custom resource and skips the update to the conflicting object. Operand updates proceed only for resources that {zero-trust-full} currently manages.
```

Author: Done

+
(link:https://issues.redhat.com/browse/SPIRE-344[SPIRE-344])

Duplicate health check port names resolved in SPIRE Server StatefulSet::
+
* Before this update, the `spire-server` and `spire-controller-manager` containers in the SPIRE Server `StatefulSet` field both declared a port named `healthz` on different container ports. Kubernetes treated this as a duplicate port name within the pod, issued a warning, and the services or probes that selected the ports by name could target the wrong container. With this release, {zero-trust-full} assigns unique port names, such as `server-healthz` and `ctrlmgr-healthz`, and updates the liveness and readiness probes to reference those names. Health checks and monitoring configurations now resolve to the intended container.
+
(link:https://issues.redhat.com/browse/SPIRE-353[SPIRE-353])

Create-only mode disabled status now updates on the main custom resource::
+

> * Before this update, after you disabled create-only mode by setting `CREATE_ONLY_MODE` to `false` in the Operator subscription, the `CreateOnlyMode` condition on the main `ZeroTrustWorkloadIdentityManager` custom resource (CR) could remain `True` with the reason `CreateOnlyModeEnabled`. Because {zero-trust-full} set that condition from operand status instead of from the `CREATE_ONLY_MODE` environment variable in the subscription, the main CR status did not reflect that create-only mode was disabled even though the operand reconciliation had resumed. With this release, {zero-trust-full} sets the `CreateOnlyMode` condition on the main custom resource directly from the `CREATE_ONLY_MODE` environment variable and updates it to `False` with reason `CreateOnlyModeDisabled` when create-only mode is turned off.

Reviewer 2: ```suggestion
* Before this update, after you disabled create-only mode by setting `CREATE_ONLY_MODE` to `false` in the Operator subscription, the `CreateOnlyMode` condition on the main `ZeroTrustWorkloadIdentityManager` custom resource (CR) could remain `True` with the reason `CreateOnlyModeEnabled`. Because {zero-trust-full} set that condition from operand status instead of from the `CREATE_ONLY_MODE` environment variable in the subscription, the main CR status did not reflect that create-only mode was disabled even though the operand reconciliation had resumed. With this release, {zero-trust-full} sets the `CreateOnlyMode` condition on the main CR directly from the `CREATE_ONLY_MODE` environment variable and updates it to `False` with reason `CreateOnlyModeDisabled` when create-only mode is turned off.
```

Author: Done

+
(link:https://issues.redhat.com/browse/SPIRE-365[SPIRE-365])

Create-only mode status updates reconcile reliably::
+

> * Before this update, {zero-trust-full} controllers wrote the CR status twice during each reconciliation cycle. The CR status was written at the start through the `SetInitialReconciliationStatus` and again at the end through the status manager. Because the informer cache could return a stale `ResourceVersion` after the first write, the deferred status update was rejected with HTTP 409 Conflict. Status conditions, including `CreateOnlyMode`, could fail to transition to the expected state even after you changed configuration in the Operator subscription. With this release, controllers apply status in a single update at the end of reconciliation, and status update retry logic now complies with the `RetryOnConflict` contract. Custom resource status updates, including create-only mode transitions, now complete reliably.

Reviewer 2: ```suggestion
* Before this update, {zero-trust-full} controllers wrote the custom resource (CR) status twice during each reconciliation cycle. The CR status was written at the start through the `SetInitialReconciliationStatus` cycle and again at the end through the status manager. Because the informer cache could return a stale `ResourceVersion` status after the first write, the deferred status update was rejected with `HTTP 409 Conflict`. Status conditions, including `CreateOnlyMode`, could fail to transition to the expected state even after you changed configuration in the Operator subscription. With this release, controllers apply the status in a single update at the end of reconciliation, and status update retry logic now complies with the `RetryOnConflict` contract. CR status updates, including create-only mode transitions, are now completed reliably.
```
For clarity, add the appropriate noun after `SetInitialReconciliationStatus` and `ResourceVersion`. I took a guess at my suggestion above.

// Module included in the following assemblies:
//
// * security/zero_trust_workload_identity_manager/zero-trust-manager-install.adoc
:_mod-docs-content-type: REFERENCE
[id="zero-trust-manager-release-notes-1-1-0_{context}"]
= {zero-trust-full} 1.1.0

Issued: 29 Jun 2026

[role="_abstract"]
This release introduces capabilities for enterprise readiness, security, and operational flexibility. The release includes SPIRE federation for cross-cluster identity, PostgreSQL support for production persistence, and enhanced security through stricter constraints and API validation.

The following advisories are available for the {zero-trust-full}:

* https://access.redhat.com/errata/RHBA-2025:23438[RHBA-2025:23438]

{zero-trust-full} supports the following components and versions:

[cols="1,1",options="header"]
|===
| Component
| Version

| SPIRE Server
| 1.13.3

| SPIRE Agent
| 1.13.3

| SPIRE Controller Manager
| 0.6.3

| SPIRE OIDC Discovery Provider
| 1.13.3

| SPIFFE CSI Driver
| 0.2.8
|===

[id="zero-trust-manager-1-1-0-features-enhancements_{context}"]
== New features and enhancements

Supported SPIFFE Helper container image::

> The Operator now provides a supported SPIFFE Helper container image for workloads that cannot use the SPIFFE Workload API directly but can read Transport Layer Security (TLS) credentials from a shared volume. The image is based on upstream SPIFFE Helper.Tthe configuration file format, command-line flags, and Workload API behavior remain compatible.

Author: Done

Key configuration points:

* Image: registry.redhat.io/zero-trust-workload-identity-manager/spiffe-helper-rhel9:<version>
* -config <path> (required)
* -daemon-mode <boolean> (default: true)
* -versions

Customer action required:

* Replace upstream images such as ghcr.io/spiffe/spiffe-helper with the {zero-trust-full} image tag that matches your installation.
* Confirm your cluster can pull from registry.redhat.io.

SPIRE UpstreamAuthority plugins for external certificate authorities::

Zero Trust Workload Identity Manager now documents SPIRE Server UpstreamAuthority plugins that obtain intermediate signing certificates from external certificate management systems while preserving Secure Production Identity Framework for Everyone (SPIFFE) identity standards.

Supported plugins:

> * cert-manager UpstreamAuthority plugin: integrates SPIRE Server with cert-manager Operator for Red Hat OpenShift* Vault UpstreamAuthority plugin: integrates SPIRE Server with the HashiCorp Vault Public Key Infrastructure (PKI) secrets engine

Author: Done

Key configuration points:

* `SpireServer.spec.upstreamAuthority.certManager`
* `SpireServer.spec.upstreamAuthority.vault`

Customer action required:

* Configure only one UpstreamAuthority plugin at a time. The `SpireServer` CR rejects configurations that specify both `certManager` and `vault`.

cert-manager UpstreamAuthority plugin::

> The cert-manager UpstreamAuthority plugin connects SPIRE Server to cert-manager Operator for Red Hat OpenShift for automated intermediate certificate provisioning. SPIRE Server creates a `CertificateRequest` custom resource, the configured Issuer or ClusterIssuer signs the request, and the SPIRE Server uses the signed

Author: Done

intermediate certificate to issue workload identities.

Key configuration points:

* `SpireServer.spec.upstreamAuthority.certManager.namespace`
* `SpireServer.spec.upstreamAuthority.certManager.issuerName`
* `SpireServer.spec.upstreamAuthority.certManager.issuerKind`
* `SpireServer.spec.upstreamAuthority.certManager.issuerGroup`

Customer action required:

> * Install cert-manager Operator for Red Hat OpenShift before configuring the plugin.

Author: Done
